mypy/binder.py

```diff
@@ -31,14 +31,11 @@ class Frame(Dict[Key, Type]):
     """
 
     def __init__(self) -> None:
+        self.types = {}  # type: Dict[Key, Type]
         self.unreachable = False
 
-
-class DeclarationsFrame(Dict[Key, Optional[Type]]):
-    """Same as above, but allowed to have None values."""
-
-    def __init__(self) -> None:
-        self.unreachable = False
+    def __getitem__(self, k: Key) -> Type:
```

Comment:
Could you remove the remaining uses of `frame[key]` into `frame.types[key]`? I notice that you did convert *some* operations on frames into operations into the `.types` attribute -- why not all? Or are there too many?